client/components/ImageService.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import axios from 'axios';
+import _ from 'lodash';
+
+import Modal from './Modal.jsx';
+import LightBox from './LightBox.jsx';
+
+import style from "../styles/style.css";
+
+class ImageService extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      didFetch: false,
+      openModal: false,
+      allImagesLoaded: false,
+      imageCount: 0,
+      curImageIndex: 0,
+      images: []
+    }
+    this.fetchNewImages = this.fetchNewImages.bind(this);
+    this.onImageLoad = this.onImageLoad.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  componentDidMount() {
+    console.log(this.props.locationId);
+    this.fetchNewImages(this.props.locationId);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.locationId !== this.props.locationId) {
+      this.fetchNewImages(this.props.locationId);
+    }
+  }
+
+  onImageLoad() {
+    let {imageCount, allImagesLoaded, images} = this.state;
+    if (imageCount === images.length) {
+      return;
+    }
+    imageCount++;
+    allImagesLoaded = imageCount === images.length;
+    this.setState({
+      imageCount: imageCount,
+      allImagesLoaded: allImagesLoaded
+    })
+  }
+
+  fetchNewImages(locationId) {
+    return axios.get(`/images/${locationId}`)
+      .then((results) => {
+        let allImagesLoaded = results.data.length === 0; //If no images are returned, then there is no point in waiting for images to load
+        this.setState({
+          images: results.data,
+          didFetch: true,
+          allImagesLoaded: allImagesLoaded
+        })
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  openModal(e) {
+    let background = document.getElementById('background');
+    let photosButton = document.getElementById('photos-button');
+    console.log(e.target)
```

Comment:
@leealbert95 Hi Albert, perhaps you can use react state rather than manipulating DOM directly by document.getElementById ? Just a suggestion!

Reply:
This part just grabs the DOM element to see if you've clicked on it, but doesn't change the elements. It might not be necessary for opening the modal (I noticed on the AirBnB page that clicking any of the buttons on the background opens up some kind of modal). However, for closeModal, I found it necessary to grab the actual DOM element because when I first made the modal, clicking anywhere within the modal's boundaries caused the modal to close (even when clicking buttons in the gallery!). It does look a little hacky to mix in direct dom access with React but it's the only way at the moment I can pick which elements in the lightbox will close the modal and which ones won't.